source/common/network/connection_impl.cc

```diff
@@ -254,51 +251,69 @@ ConnectionImpl::PostIoAction ConnectionImpl::doReadFromSocket() {
 
     // Remote close. Might need to raise data before raising close.
     if (rc == 0) {
-      return PostIoAction::Close;
+      action = PostIoAction::Close;
+      break;
     }
 
     // Remote error (might be no data).
     if (rc == -1) {
```

Comment:
minor nit:  this code would be easier to read if this was an 'else if' off of the previous if